main.go

```diff
@@ -83,17 +88,24 @@ func init() {
 	_ = azurev1beta1.AddToScheme(scheme)
 	_ = azurev1alpha2.AddToScheme(scheme)
 	// +kubebuilder:scaffold:scheme
+
+	// We need to query AAD identity types, so add them to the scheme
+	aadPodIdentityGroupVersion := schema.GroupVersion{Group: aadpodv1.CRDGroup, Version: aadpodv1.CRDVersion}
+	scheme.AddKnownTypes(aadPodIdentityGroupVersion,
+		&aadpodv1.AzureIdentity{},
+		&aadpodv1.AzureIdentityList{},
+		&aadpodv1.AzureIdentityBinding{},
+		&aadpodv1.AzureIdentityBindingList{},
+		&aadpodv1.AzureAssignedIdentity{},
+		&aadpodv1.AzureAssignedIdentityList{},
+		&aadpodv1.AzurePodIdentityException{},
+		&aadpodv1.AzurePodIdentityExceptionList{})
+	metav1.AddToGroupVersion(scheme, aadPodIdentityGroupVersion)
 }
 
 // +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=core,resources=events,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=azure.microsoft.com,resources=events,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups=azure.microsoft.com,resources=azuresqlusers,verbs=get;list;watch;create;update;patch;delete
```

Comment:
Because I moved these to follow the pattern of all other resources, where the RBAC rules are defined alongside the resource.

They were actually defined in _both_ places but Kubebuilder wasn't detecting the ones alongside the resource because of extremely awkward spacing requirements. See for example `mysqluser_controller.go`, where the comment before the `kubebuilder` comment made `kubebuilder` not detect it.

Reply:
Just went through and found a controller-tools bug: https://github.com/kubernetes-sigs/controller-tools/pull/437

So basically we had these duplicated but only one was working.

Reply:
Makes sense, thanks!